File: Classes/IntroViewController.m
//
//  IntroViewController.m
//  iSub
//
//  Created by Ben Baron on 1/27/11.
//  Copyright 2011 Ben Baron. All rights reserved.
//

#import "iSub-Swift.h"
#import "IntroViewController.h"
#import "Imports.h"
#import "ServerListViewController.h"
#import <MediaPlayer/MediaPlayer.h>

@implementation IntroViewController
@synthesize introVideo, testServer, ownServer, sunkenLogo;

- (BOOL)prefersStatusBarHidden
{
    return YES;
}

- (BOOL)shouldAutorotate
{
    if (settingsS.isRotationLockEnabled && [UIDevice currentDevice].orientation != UIDeviceOrientationPortrait)
        return NO;
    
    return YES;
}

- (void)dealloc
{
	[[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (void)didReceiveMemoryWarning 
{
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc. that aren't in use.
}

- (void)viewDidLoad
{
	[super viewDidLoad];
	
	if (IS_IPAD())
		sunkenLogo.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleRightMargin | UIViewAutoresizingFlexibleBottomMargin;
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
	[[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(dismissFast) name:ISMSNotification_EnteringOfflineMode object:nil];
}

- (void)viewWillDisappear:(BOOL)animated
{
	[[NSNotificationCenter defaultCenter] removeObserver:self name:ISMSNotification_EnteringOfflineMode object:nil];
}

- (void)dismissFast
{
	[self dismissViewControllerAnimated:NO completion:nil];
}

- (IBAction)buttonPress:(id)sender
{
	if (sender == self.introVideo)
	{
		NSURL *introUrl = nil;
		if (IS_IPAD())
			introUrl = [NSURL URLWithString:@"http://isubapp.com/intro/ipad/prog_index.m3u8"];
		else if (SCREEN_SCALE() == 2.0)
			introUrl = [NSURL URLWithString:@"http://isubapp.com/intro/iphone4/prog_index.m3u8"];
		else
			introUrl = [NSURL URLWithString:@"http://isubapp.com/intro/iphone/prog_index.m3u8"];
		
		if ([MPMoviePlayerController instancesRespondToSelector:@selector(view)]) 
		{
			// Running on 3.2+
			MPMoviePlayerViewController *moviePlayer = [[MPMoviePlayerViewController alloc] initWithContentURL:introUrl];
			// Assuming self is a UIViewController
			[self presentMoviePlayerViewControllerAnimated:moviePlayer];
			// This line might be needed
			[moviePlayer.moviePlayer play];
			
			// TODO, 
		} 
		else 
		{
			MPMoviePlayerController *moviePlayer= [[MPMoviePlayerController alloc] initWithContentURL:introUrl];
			[moviePlayer play];
		}	
	}
	else if (sender == self.testServer)
	{
		[self dismissViewControllerAnimated:YES completion:nil];
	}
	else if (sender == self.ownServer)
	{
        [self dismissViewControllerAnimated:NO completion:nil];
        
        [appDelegateS showSettings];
	}
}

@end

File: Classes/Old UI/IntroViewController.swift
//
//  IntroViewController.swift
//  iSub Beta
//
//  Created by Benjamin Baron on 7/2/17.
//  Copyright © 2017 Ben Baron. All rights reserved.
//

import Foundation
import AVKit

class IntroViewController: UIViewController {
    @IBOutlet weak var introVideo: UIButton!
    @IBOutlet weak var testServer: UIButton!
    @IBOutlet weak var ownServer: UIButton!
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var shouldAutorotate: Bool {
        return true
    }
    
    @IBAction func buttonPress(_ sender: UIButton) {
        if sender == introVideo {
            let introUrl: URL?
            if UIScreen.main.scale > 1.0 {
                introUrl = URL(string: "http://isubapp.com/intro/iphone4/prog_index.m3u8")
            } else {
                introUrl = URL(string: "http://isubapp.com/intro/iphone/prog_index.m3u8")
            }
            
            if let introUrl = introUrl {
                let playerViewController = AVPlayerViewController()
                playerViewController.player = AVPlayer(url: introUrl)
                self.present(playerViewController, animated: true) {
                    playerViewController.player?.play()
                }
            }
        } else if sender == testServer {
            self.dismiss(animated: true, completion: nil)
        } else if sender == ownServer {
            self.dismiss(animated: true, completion: nil)
            if let menu = AppDelegate.si.sidePanelController.leftPanel as? MenuViewController {
                menu.showSettings()
            }
        }
    }
}
